app/Http/Controllers/Teams/TeamsController.php

```diff
@@ -70,61 +73,33 @@ public function inactivateTeam()
      *
      * @return array
      */
-    public function create (Request $request)
+    public function create (CreateTeamRequest $request, CreateTeamAction $action)
     {
-        $request->validate([
-            'name' => 'required|min:3|max:100|unique:teams',
-            'identifier' => 'required|min:3|max:15|unique:teams',
-            'teamType' => 'required'
-        ]);
-
+        /** @var User $user */
         $user = Auth::user();
 
         if ($user->remaining_teams === 0) return ['success' => false, 'msg' => 'max-created'];
 
-        $team = Team::create([
-            'created_by' => $user->id,
-            'name' => $request->name,
-            'type_id' => $request->teamType,
-            'leader' => $user->id,
-            'identifier' => $request->identifier
-        ]);
-
-        // Broadcast live event to the global map
-        event (new TeamCreated($team->name));
-
-        // Have the user join this team
-        $user->teams()->attach($team);
-
-        $user->active_team = $team->id;
-        $user->remaining_teams--;
-        $user->save();
+        $team = $action->run($user, $request->all());
```

Comment:
It's not clear what `$action` is. Maybe we should use the name `$createTeamAction` ?

Reply:
The action's purpose is in its type, I don't think it's necessary to use a fully descriptive name here. I kept it short so the method's signature stays in one line.